### PR TITLE
feat(rest-api): when creating audio, allow params for episode/audio publication

### PR DESCRIPTION
**Pull request by eteubert**
Via discussion in Slack: make it easier for API consumer to create an audio with episode / audio publication by taking parameters for the episode / audio publication while creating the audio.

/cc @Fischaela

### CODE REVIEW

File: lib/radiator_web/controllers/api/audio_controller.ex [hunk 1]
```diff
@@ -3,21 +3,33 @@ defmodule RadiatorWeb.Api.AudioController do
 
   alias Radiator.Directory.Editor
 
-  def create(conn, %{"network_id" => network_id, "audio" => params}) do
+  def create(conn, params = %{"network_id" => network_id}) do
```
Comment (monkeydom) [inline lib/radiator_web/controllers/api/audio_controller.ex]: Either documentation is wrong, or you are matching wrong. (the "network_id" key is still inside an "audio")

File: lib/radiator_web/controllers/api/audio_controller.ex [hunk 2]
```diff
@@ -3,21 +3,33 @@ defmodule RadiatorWeb.Api.AudioController do
...
+           ) do
       conn
       |> put_status(:created)
       |> put_resp_header("location", Routes.api_audio_path(conn, :show, audio))
       |> render("show.json", %{audio: audio})
     end
   end
 
-  def create(conn, %{"episode_id" => episode_id, "audio" => params}) do
+  def create(conn, params = %{"episode_id" => episode_id}) do
```
Comment (monkeydom) [inline lib/radiator_web/controllers/api/audio_controller.ex]: dito. Maybe a test added would be good?

File: lib/radiator_web/controllers/api/audio_controller.ex (hunk 1 above)
Comment (eteubert) [reply lib/radiator_web/controllers/api/audio_controller.ex]: These (`network_id` and `episode_id`) are not params supplied but the client but the ones extracted from the route. Comes from the nesting under network/episode resource.

Comment (monkeydom) [reply lib/radiator_web/controllers/api/audio_controller.ex]: Ah, some magic.

**Pull request merged**